pkg/ema.py

```diff
@@ -131,21 +134,23 @@ def poll_ema(id, empathid, action_idx, duration=300, freq=5):
 def setup_message(message, type='binary'):
   suid = '22'
   template_path = os.path.join(DIR_PATH, 'message_templates', f'{type}.bin')
+
   with open(template_path, 'rb') as f:
     template = f.read().decode()
 
-  ema_survey = re.sub(r's:(\d+):(\[\[MESSAGE_PLACEHOLDEER\]\])', r's:' + len(message.encode()) + ':' + message, template)
+
+  ema_survey = re.sub(r's:(\d+):(\[\[MESSAGE_PLACEHOLDEER\]\])', r's:' + str(len(message.encode())) + ':' + message, template)
   buf = zlib.compress(ema_survey.encode())
 
   try:
     conn = get_conn()
     with conn.cursor() as cursor:
       # Create a new record
-      sql = "UPDATE ema_context SET variables = %s WHERE suid = %s"
```

Comment:
oh, damn, this space really matters?
when i was writing this, i was wondering how the query can know if the space if a part of the buffer

Reply:
Oh no, it works with or without the space. I was double checking and I forgot to delete the spaces afterwards.